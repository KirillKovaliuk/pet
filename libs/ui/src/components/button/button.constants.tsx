export const BUTTON_ICON_SIZES = {
  MEDIUM: 20,
  LARGE: 24,
} as const;

export const HIERARCHIES = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  TERTIARY: 'tertiary',
} as const;

export const BUTTON_TYPE = 'button';
export const SUBMIT_TYPE = 'submit';
export const RESET_TYPE = 'reset';

export const BUTTON_TYPES = {
  BUTTON_TYPE,
  SUBMIT_TYPE,
  RESET_TYPE,
} as const;
