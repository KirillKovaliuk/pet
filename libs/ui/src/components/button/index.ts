export * from './button';
export * from './button.declarations';
export * from './button.styles';
export * from './button.constants';
export * from './button.utils';
