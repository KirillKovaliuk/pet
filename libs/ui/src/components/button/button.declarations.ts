import { ReactNode, SyntheticEvent } from 'react';

import { TSize } from '../../types';
import { TIconName } from '../icon';

import { BUTTON_TYPES, HIERARCHIES } from './button.constants';

type THierarchyKeys = keyof typeof HIERARCHIES;
export type THierarchy = typeof HIERARCHIES[THierarchyKeys];

type TButtonKeys = keyof typeof BUTTON_TYPES;
export type TButton = typeof BUTTON_TYPES[TButtonKeys];

export interface IButtonProps {
  children: ReactNode;
  onClick?: (event: SyntheticEvent) => void;
  className?: string;
  size?: TSize;
  iconBefore?: TIconName;
  iconAfter?: TIconName;
  disabled?: boolean;
  destructive?: boolean;
  colored?: boolean;
  hierarchy?: THierarchy;
  type?: TButton;
  fullWidth?: boolean;
}

type TExcludedMainProps = 'destructive' | 'colored' | 'hierarchy' | 'size' | 'fullWidth';

export interface IButtonStyles extends Omit<IButtonProps, TExcludedMainProps> {
  $destructive?: boolean;
  $colored?: boolean;
  $hierarchy?: THierarchy;
  $size?: TSize;
  $fullWidth?: boolean;
}
