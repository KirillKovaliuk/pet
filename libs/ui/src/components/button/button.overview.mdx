import { Meta } from '@storybook/addon-docs';
import { List } from "../list";
import { SIZES } from "../../constants";
import { Button } from "./button";
import { Icon } from "../icon";

<Meta title="Elements/button" />

# Кнопка

Кнопка - это интерактивный элемент, который приводит к описанному в нем действию. Они являются одним
из самых важных компонентов в современном дизайне пользовательского интерфейса, поэтому очень важно
правильно использовать их в своих проектах. Самое важное правило при разработке кнопки - она должна
выглядеть как кнопка. Она должна быть очевидной. Пользователи знакомы с тем, как выглядят кнопки и
как они себя ведут. Чем больше кнопка "похожа" на кнопку, тем легче ее быстро и подсознательно обнаружить.

Когда на странице есть несколько действий, важно расположить их в четкой иерархии. Каждое действие на
странице занимает определенное место в иерархии важности. Большинство компонентов и страниц имеют только
одно "основное" действие. Это главное действие, которое, скорее всего, совершит пользователь (или которое
дизайнер хочет, чтобы он совершил). Другие действия являются "вторичными". На странице может быть более
одного второстепенного действия. Как правило, их следует различать и по возможности располагать в последовательной
иерархии. Возможно, вам даже понадобится третье "третичное" или даже четвертое "четвертичное" действие. Стиль
кнопки в первую очередь используется для того, чтобы отличить более важные действия от менее важных - первичные
действия должны быть вариантом по умолчанию и наиболее очевидными.

Бытует мнение, что все "негативные" или "разрушительные" действия должны быть выделены большой красной кнопкой.
В большинстве случаев это не обязательно, и дизайн может выглядеть и ощущаться перегруженным. Тем не менее,
если конкретное "разрушительное" действие является основным, обычно полезно визуально обозначить его "разрушительным"
вариантом кнопки. Это дает пользователю понять, что он собирается совершить действие, имеющее серьезные последствия.

<Icon name="socials-figma" size={14} /> [Дизайн](https://www.figma.com/file/qXNcgpuRBpzUNmXAyhWOWN/%E2%9D%96-Untitled-UI-%E2%80%93-PRO-VARIABLES-(v4.0)-qkXWNpVGpSCF-(Copy)?type=design&node-id=3287-429411)

Ниже вы можете увидеть основные варианты кнопок:

## Primary(Главная)

#### Default

<List data={Object.values(SIZES)} horizontal gap={12}>
  {(size) => (
    <Button
      iconBefore="placeholder"
      iconAfter="placeholder"
      size={size}
    >
      Button CTA
    </Button>
  )}
</List>

##### Destructive

<List data={Object.values(SIZES)} horizontal gap={12}>
  {(size) => (
    <Button
      iconBefore="placeholder"
      iconAfter="placeholder"
      size={size}
      destructive
    >
      Button CTA
    </Button>
  )}
</List>

##### Disabled

<List data={Object.values(SIZES)} horizontal gap={12}>
  {(size) => (
    <Button
      iconBefore="placeholder"
      iconAfter="placeholder"
      size={size}
      disabled
    >
      Button CTA
    </Button>
  )}
</List>


## Secondary(Вторичная)

#### Default

<List data={Object.values(SIZES)} horizontal gap={12}>
  {(size) => (
    <Button
      hierarchy="secondary"
      iconBefore="placeholder"
      iconAfter="placeholder"
      size={size}
    >
      Button CTA
    </Button>
  )}
</List>

#### Colored

<List data={Object.values(SIZES)} horizontal gap={12}>
  {(size) => (
    <Button
      hierarchy="secondary"
      iconBefore="placeholder"
      iconAfter="placeholder"
      size={size}
      colored
    >
      Button CTA
    </Button>
  )}
</List>


##### Destructive

<List data={Object.values(SIZES)} horizontal gap={12}>
  {(size) => (
    <Button
      hierarchy="secondary"
      iconBefore="placeholder"
      iconAfter="placeholder"
      size={size}
      destructive
    >
      Button CTA
    </Button>
  )}
</List>

##### Disabled

<List data={Object.values(SIZES)} horizontal gap={12}>
  {(size) => (
    <Button
      hierarchy="secondary"
      iconBefore="placeholder"
      iconAfter="placeholder"
      size={size}
      disabled
    >
      Button CTA
    </Button>
  )}
</List>


## Tertiary(Третичная)

#### Default

<List data={Object.values(SIZES)} horizontal gap={12}>
  {(size) => (
    <Button
      hierarchy="tertiary"
      iconBefore="placeholder"
      iconAfter="placeholder"
      size={size}
    >
      Button CTA
    </Button>
  )}
</List>

#### Colored

<List data={Object.values(SIZES)} horizontal gap={12}>
  {(size) => (
    <Button
      hierarchy="tertiary"
      iconBefore="placeholder"
      iconAfter="placeholder"
      size={size}
      colored
    >
      Button CTA
    </Button>
  )}
</List>


##### Destructive

<List data={Object.values(SIZES)} horizontal gap={12}>
  {(size) => (
    <Button
      hierarchy="tertiary"
      iconBefore="placeholder"
      iconAfter="placeholder"
      size={size}
      destructive
    >
      Button CTA
    </Button>
  )}
</List>

##### Disabled

<List data={Object.values(SIZES)} horizontal gap={12}>
  {(size) => (
    <Button
      hierarchy="tertiary"
      iconBefore="placeholder"
      iconAfter="placeholder"
      size={size}
      disabled
    >
      Button CTA
    </Button>
  )}
</List>
