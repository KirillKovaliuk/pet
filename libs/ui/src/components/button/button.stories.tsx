/* eslint-disable @typescript-eslint/no-explicit-any */
import { SIZES } from '../../constants';
import {
  ArgumentTypeBuilder,
  booleanArgumentType,
  childrenArgumentType,
  hierarchyArgumentType,
  optionalIconArgumentType,
  sizeArgumentType,
} from '../../utils';

import { Button as ButtonUI } from './button';
import { HIERARCHIES } from './button.constants';

export function Sandbox(props: any) {
  return (
    <ButtonUI {...props} />
  );
}

export default {
  title: 'Elements/Button',
  component: Sandbox,
  argTypes: {
    children: ArgumentTypeBuilder
      .from(childrenArgumentType)
      .setDescription('Button content')
      .setRequired(true)
      .setDefaultValue('-'),
    hierarchy: hierarchyArgumentType,
    size: sizeArgumentType,
    destructive: ArgumentTypeBuilder
      .from(booleanArgumentType)
      .setDescription('Is action destructive?'),
    colored: ArgumentTypeBuilder
      .from(booleanArgumentType)
      .setDescription('Is button colored? Ignored when primary'),
    disabled: ArgumentTypeBuilder
      .from(booleanArgumentType)
      .setDescription('Is button unclickable?'),
    iconBefore: optionalIconArgumentType,
    iconAfter: optionalIconArgumentType,
    fullWidth: ArgumentTypeBuilder
      .from(booleanArgumentType)
      .setDescription('Is button full width?'),
  },
  args: {
    children: 'label',
    hierarchy: HIERARCHIES.PRIMARY,
    size: SIZES.MEDIUM,
    destructive: false,
    colored: false,
    disabled: false,
    fullWidth: false,
  },
};
