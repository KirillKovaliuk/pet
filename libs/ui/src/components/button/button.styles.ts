import styled, { css } from 'styled-components';

import { PALETTE, SIZES } from '../../constants';

import { HIERARCHIES } from './button.constants';
import { IButtonStyles, THierarchy } from './button.declarations';

import {
  largeTextStyles,
  mediumTextStyles,
  smallTextStyles,
} from '../typography/typography.styles';

// sizes declaration start
const extraSmallButtonStyles = css`
  min-height: 36px;
  padding: ${({ theme }) => theme.spacing.md} ${({ theme }) => theme.spacing.lg};
  gap: ${({ theme }) => theme.spacing.xs};
  ${smallTextStyles};
`;

const smallButtonStyles = css`
  min-height: 40px;
  padding: 10px 14px;
  gap: ${({ theme }) => theme.spacing.xs};
  ${smallTextStyles};
`;

const mediumButtonStyles = css`
  min-height: 44px;
  padding: 10px ${({ theme }) => theme.spacing.xl};
  gap: ${({ theme }) => theme.spacing.sm};
  ${mediumTextStyles};
`;

const largeButtonStyles = css`
  min-height: 48px;
  padding: ${({ theme }) => theme.spacing.lg} 18px;
  gap: ${({ theme }) => theme.spacing.sm};
  ${mediumTextStyles};
`;

const extraLargeButtonStyles = css`
  min-height: 60px;
  padding: ${({ theme }) => theme.spacing.xl} 22px;
  gap: ${({ theme }) => theme.spacing.lg};
  ${largeTextStyles};
`;

const BUTTON_SIZE_STYLES_MAP = {
  [SIZES.EXTRA_SMALL]: extraSmallButtonStyles,
  [SIZES.SMALL]: smallButtonStyles,
  [SIZES.MEDIUM]: mediumButtonStyles,
  [SIZES.LARGE]: largeButtonStyles,
  [SIZES.EXTRA_LARGE]: extraLargeButtonStyles,
};

const buttonSizeStyles = css<IButtonStyles>`
  ${({ $size }) => BUTTON_SIZE_STYLES_MAP[$size || SIZES.MEDIUM]};
`;

// sizes declaration end

const getButtonStyledType = ($destructive?: boolean, $colored?: boolean) => {
  let type = 'default';

  if ($destructive) {
    type = 'error';
  } else if ($colored) {
    type = 'colored';
  }

  return type;
};

const getButtonStylesByType = (hierarchy: THierarchy, propertyName: string) => css<IButtonStyles>`
  ${({ theme, $destructive, $colored }) => {
    let type = getButtonStyledType($destructive, $colored);

    if (hierarchy === 'primary' && $colored) {
      type = $destructive ? 'error' : 'default';
    }

    // @TODO: improve types
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    return theme.button[hierarchy][type][propertyName];
  }}
`;

const getPrimaryFocusRing = css<IButtonStyles>`${({ $destructive, theme }) => {
  if ($destructive) {
    return theme.ring['error-shadow-xs'];
  }

  return theme.ring['brand-shadow-xs'];
}}`;

const getSecondaryFocusRing = css<IButtonStyles>`${({ $destructive, $colored, theme }) => {
  if ($destructive) {
    return theme.ring['error-shadow-xs'];
  } if ($colored) {
    return theme.ring['brand-shadow-xs'];
  }

  return theme.ring['gray-shadow-xs'];
}}`;

// primary button styles declaration start
const primaryButtonStyles = css<IButtonStyles>`
  background: ${getButtonStylesByType('primary', 'background')};
  border: 1px solid ${getButtonStylesByType('primary', 'border')};
  color: ${getButtonStylesByType('primary', 'foreground')};
  box-shadow: ${({ theme }) => theme.shadows['extra-small']};

  &:hover:not(:disabled) {
    background: ${getButtonStylesByType('primary', 'background-hover')};
    border: 1px solid ${getButtonStylesByType('primary', 'border-hover')};
  }

  &:focus {
    box-shadow: ${getPrimaryFocusRing};
  }

  &:disabled {
    color: ${({ theme }) => theme.foreground.disabled};
    background: ${({ theme }) => theme.background.disabled};
    border: 1px solid ${({ theme }) => theme.border['disabled-subtle']};
    cursor: not-allowed;
  }
`;
// primary button styles declaration end

// secondary button styles declaration start
const secondaryButtonStyles = css<IButtonStyles>`
  background: ${getButtonStylesByType('secondary', 'background')};
  border: 1px solid ${getButtonStylesByType('secondary', 'border')};
  color: ${getButtonStylesByType('secondary', 'foreground')};

  &:hover:not(:disabled) {
    background: ${getButtonStylesByType('secondary', 'background-hover')};
    color: ${getButtonStylesByType('secondary', 'foreground-hover')};
    border: 1px solid ${getButtonStylesByType('secondary', 'border-hover')};
  }

  &:focus {
    box-shadow: ${getSecondaryFocusRing};
  }

  &:disabled {
    color: ${({ theme }) => theme.foreground.disabled};
    background: ${({ theme }) => theme.background.primary};
    border: 1px solid ${({ theme }) => theme.border['disabled-subtle']};
    cursor: not-allowed;
  }
`;
// secondary button styles declaration end

// tertiary button styles declaration start
const tertiaryButtonStyles = css<IButtonStyles>`
  color: ${getButtonStylesByType('tertiary', 'foreground')};
  border: none;
  background: none;
  box-shadow: none;
  border: 1px solid ${PALETTE.base.transparent};

  &:hover:not(:disabled) {
    color: ${getButtonStylesByType('tertiary', 'foreground-hover')};
    background: ${getButtonStylesByType('tertiary', 'background-hover')};
  }

  &:disabled {
    color: ${({ theme }) => theme.foreground.disabled};
    border: 1px solid ${PALETTE.base.transparent};
  }
`;
// tertiary button styles declaration end

const BUTTON_HIERARCHY_STYLES_MAP = {
  [HIERARCHIES.PRIMARY]: primaryButtonStyles,
  [HIERARCHIES.SECONDARY]: secondaryButtonStyles,
  [HIERARCHIES.TERTIARY]: tertiaryButtonStyles,
};

const buttonHierarchyStyles = css<IButtonStyles>`
  ${({ $hierarchy }) => {
    const safeHierarchy = $hierarchy || HIERARCHIES.PRIMARY;

    return BUTTON_HIERARCHY_STYLES_MAP[safeHierarchy];
  }}
`;

export const StyledButton = styled.button<IButtonStyles>`
  display: inline-flex;
  justify-content: center;
  align-items: center;
  border-radius: 8px;
  cursor: pointer;
  width: ${({ $fullWidth }) => ($fullWidth ? '100%' : 'auto')};
  transition: all .2s ease-in;
  font-weight: 600;
  font-style: normal;

  &:focus {
    outline: none;
  }

  &:disabled {
    cursor: not-allowed;
    user-select: none;
  }

  ${buttonSizeStyles};
  ${buttonHierarchyStyles};
`;
