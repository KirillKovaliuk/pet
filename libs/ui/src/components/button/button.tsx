'use client';

import { forwardRef } from 'react';

import { Icon } from '../icon';

import { BUTTON_TYPES } from './button.constants';
import { IButtonProps } from './button.declarations';
import { getIconSize } from './button.utils';

import { StyledButton } from './button.styles';

export const Button = (
  forwardRef<HTMLButtonElement, IButtonProps>(
    ({
      type = BUTTON_TYPES.BUTTON_TYPE,
      children,
      className,
      onClick,
      size,
      hierarchy,
      iconBefore,
      iconAfter,
      destructive,
      colored,
      disabled,
      fullWidth,
    }, ref) => {
      const iconSize = getIconSize(size);

      return (
        <StyledButton
          ref={ref}
          onClick={onClick}
          disabled={disabled}
          className={className}
          type={type}
          $size={size}
          $hierarchy={hierarchy}
          $destructive={destructive}
          $colored={colored}
          $fullWidth={fullWidth}
        >
          { iconBefore && <Icon name={iconBefore} size={iconSize} />}
          {children}
          {iconAfter && <Icon name={iconAfter} size={iconSize} />}
        </StyledButton>
      );
    },
  )
);
