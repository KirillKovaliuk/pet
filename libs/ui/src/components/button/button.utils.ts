import { SIZES } from '../../constants';
import { TSize } from '../../types';

import { BUTTON_ICON_SIZES } from './button.constants';

export const getIconSize = (size?: TSize) => (
  size === SIZES.EXTRA_LARGE
    ? BUTTON_ICON_SIZES.LARGE
    : BUTTON_ICON_SIZES.MEDIUM
);
