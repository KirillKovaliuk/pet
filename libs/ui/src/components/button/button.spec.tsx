import renderer, { act } from 'react-test-renderer';

import { Theme } from '../theme';

import { Button } from './button';

describe('<Button />:', () => {
  it('should render primary button component correctly', () => {
    const container = renderer
      .create(
        <Theme>
          <Button>primary label</Button>
        </Theme>,
      );

    expect(container.toJSON()).toMatchSnapshot();
  });

  it('should render primary button component with icons correctly', () => {
    const container = renderer
      .create(
        <Theme>
          <Button iconAfter="edit" iconBefore="check">primary label</Button>
        </Theme>,
      );

    expect(container.toJSON()).toMatchSnapshot();
  });

  it('should render primary fullwidth button component correctly', () => {
    const container = renderer
      .create(
        <Theme>
          <Button fullWidth>primary label</Button>
        </Theme>,
      );

    expect(container.toJSON()).toMatchSnapshot();
  });

  it('should render primary destructive button component correctly', () => {
    const container = renderer
      .create(
        <Theme>
          <Button destructive>primary label</Button>
        </Theme>,
      );

    expect(container.toJSON()).toMatchSnapshot();
  });

  it('should render secondary button component correctly', () => {
    const container = renderer
      .create(
        <Theme>
          <Button hierarchy="secondary">secondary label</Button>
        </Theme>,
      );

    expect(container.toJSON()).toMatchSnapshot();
  });

  it('should render secondary colored button component correctly', () => {
    const container = renderer
      .create(
        <Theme>
          <Button hierarchy="secondary" colored>secondary label</Button>
        </Theme>,
      );

    expect(container.toJSON()).toMatchSnapshot();
  });

  it('should render tertiary button component correctly', () => {
    const container = renderer
      .create(
        <Theme>
          <Button hierarchy="tertiary">tertiary label</Button>
        </Theme>,
      );

    expect(container.toJSON()).toMatchSnapshot();
  });

  it('should render tertiary colored button component correctly', () => {
    const container = renderer
      .create(
        <Theme>
          <Button hierarchy="tertiary" colored>tertiary label</Button>
        </Theme>,
      );

    expect(container.toJSON()).toMatchSnapshot();
  });

  it('should trigger click event correctly', () => {
    const mockFunction = jest.fn();

    const container = renderer
      .create(
        <Theme>
          <Button
            hierarchy="tertiary"
            onClick={mockFunction}
            colored
          >
            tertiary label
          </Button>
        </Theme>,
      );

    act(() => {
      container.root.findByType('button').props.onClick();
    });

    expect(mockFunction).toHaveBeenCalled();
  });
});
