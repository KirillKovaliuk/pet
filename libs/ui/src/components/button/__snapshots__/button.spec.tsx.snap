// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button />: should render primary button component correctly 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 8px;
  cursor: pointer;
  width: auto;
  -webkit-transition: all .2s ease-in;
  transition: all .2s ease-in;
  font-weight: 600;
  font-style: normal;
  min-height: 44px;
  padding: 10px 16px;
  gap: 6px;
  font-size: 16px;
  line-height: 24px;
  background: #7F56D9;
  border: 1px solid  #7F56D9;
  color: #ffffff;
  box-shadow: 0px 1px 2px rgba(16,24,40,0.05);
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:hover:not(:disabled) {
  background: #9E77ED;
  border: 1px solid  #9E77ED;
}

.c0:focus {
  box-shadow: 0px 1px 2px 0px rgba(22,27,38,0.05),0px 0px 0px 4px rgba(158,119,237,0.24);
}

.c0:disabled {
  color: #85888E;
  background: #1F242F;
  border: 1px solid #1F242F;
  cursor: not-allowed;
}

<button
  className="c0"
  type="button"
>
  primary label
</button>
`;

exports[`<Button />: should render primary button component with icons correctly 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
  height: 20px;
}

.c1 svg {
  width: 20px;
  height: 20px;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 8px;
  cursor: pointer;
  width: auto;
  -webkit-transition: all .2s ease-in;
  transition: all .2s ease-in;
  font-weight: 600;
  font-style: normal;
  min-height: 44px;
  padding: 10px 16px;
  gap: 6px;
  font-size: 16px;
  line-height: 24px;
  background: #7F56D9;
  border: 1px solid  #7F56D9;
  color: #ffffff;
  box-shadow: 0px 1px 2px rgba(16,24,40,0.05);
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:hover:not(:disabled) {
  background: #9E77ED;
  border: 1px solid  #9E77ED;
}

.c0:focus {
  box-shadow: 0px 1px 2px 0px rgba(22,27,38,0.05),0px 0px 0px 4px rgba(158,119,237,0.24);
}

.c0:disabled {
  color: #85888E;
  background: #1F242F;
  border: 1px solid #1F242F;
  cursor: not-allowed;
}

<button
  className="c0"
  type="button"
>
  <span
    className="c1"
  >
    <check.svg />
  </span>
  primary label
  <span
    className="c1"
  >
    <edit.svg />
  </span>
</button>
`;

exports[`<Button />: should render primary destructive button component correctly 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 8px;
  cursor: pointer;
  width: auto;
  -webkit-transition: all .2s ease-in;
  transition: all .2s ease-in;
  font-weight: 600;
  font-style: normal;
  min-height: 44px;
  padding: 10px 16px;
  gap: 6px;
  font-size: 16px;
  line-height: 24px;
  background: #D92D20;
  border: 1px solid  #D92D20;
  color: #ffffff;
  box-shadow: 0px 1px 2px rgba(16,24,40,0.05);
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:hover:not(:disabled) {
  background: #F04438;
  border: 1px solid  #F04438;
}

.c0:focus {
  box-shadow: 0px 1px 2px 0px rgba(22,27,38,0.05),0px 0px 0px 4px rgba(240,68,56,0.24);
}

.c0:disabled {
  color: #85888E;
  background: #1F242F;
  border: 1px solid #1F242F;
  cursor: not-allowed;
}

<button
  className="c0"
  type="button"
>
  primary label
</button>
`;

exports[`<Button />: should render primary fullwidth button component correctly 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 8px;
  cursor: pointer;
  width: 100%;
  -webkit-transition: all .2s ease-in;
  transition: all .2s ease-in;
  font-weight: 600;
  font-style: normal;
  min-height: 44px;
  padding: 10px 16px;
  gap: 6px;
  font-size: 16px;
  line-height: 24px;
  background: #7F56D9;
  border: 1px solid  #7F56D9;
  color: #ffffff;
  box-shadow: 0px 1px 2px rgba(16,24,40,0.05);
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:hover:not(:disabled) {
  background: #9E77ED;
  border: 1px solid  #9E77ED;
}

.c0:focus {
  box-shadow: 0px 1px 2px 0px rgba(22,27,38,0.05),0px 0px 0px 4px rgba(158,119,237,0.24);
}

.c0:disabled {
  color: #85888E;
  background: #1F242F;
  border: 1px solid #1F242F;
  cursor: not-allowed;
}

<button
  className="c0"
  type="button"
>
  primary label
</button>
`;

exports[`<Button />: should render secondary button component correctly 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 8px;
  cursor: pointer;
  width: auto;
  -webkit-transition: all .2s ease-in;
  transition: all .2s ease-in;
  font-weight: 600;
  font-style: normal;
  min-height: 44px;
  padding: 10px 16px;
  gap: 6px;
  font-size: 16px;
  line-height: 24px;
  background: #161B26;
  border: 1px solid  #333741;
  color: #CECFD2;
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:hover:not(:disabled) {
  background: #1F242F;
  color: #F0F1F1;
  border: 1px solid  #333741;
}

.c0:focus {
  box-shadow: 0px 1px 2px 0px rgba(22,27,38,0.05),0px 0px 0px 4px rgba(148,150,156,0.14);
}

.c0:disabled {
  color: #85888E;
  background: #0C111D;
  border: 1px solid #1F242F;
  cursor: not-allowed;
}

<button
  className="c0"
  type="button"
>
  secondary label
</button>
`;

exports[`<Button />: should render secondary colored button component correctly 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 8px;
  cursor: pointer;
  width: auto;
  -webkit-transition: all .2s ease-in;
  transition: all .2s ease-in;
  font-weight: 600;
  font-style: normal;
  min-height: 44px;
  padding: 10px 16px;
  gap: 6px;
  font-size: 16px;
  line-height: 24px;
  background: #161B26;
  border: 1px solid  #333741;
  color: #CECFD2;
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:hover:not(:disabled) {
  background: #1F242F;
  color: #F0F1F1;
  border: 1px solid  #333741;
}

.c0:focus {
  box-shadow: 0px 1px 2px 0px rgba(22,27,38,0.05),0px 0px 0px 4px rgba(158,119,237,0.24);
}

.c0:disabled {
  color: #85888E;
  background: #0C111D;
  border: 1px solid #1F242F;
  cursor: not-allowed;
}

<button
  className="c0"
  type="button"
>
  secondary label
</button>
`;

exports[`<Button />: should render tertiary button component correctly 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 8px;
  cursor: pointer;
  width: auto;
  -webkit-transition: all .2s ease-in;
  transition: all .2s ease-in;
  font-weight: 600;
  font-style: normal;
  min-height: 44px;
  padding: 10px 16px;
  gap: 6px;
  font-size: 16px;
  line-height: 24px;
  color: #94969C;
  border: none;
  background: none;
  box-shadow: none;
  border: 1px solid rgba(0,0,0,0);
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:hover:not(:disabled) {
  color: #ECECED;
  background: #1F242F;
}

.c0:disabled {
  color: #85888E;
  border: 1px solid rgba(0,0,0,0);
}

<button
  className="c0"
  type="button"
>
  tertiary label
</button>
`;

exports[`<Button />: should render tertiary colored button component correctly 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 8px;
  cursor: pointer;
  width: auto;
  -webkit-transition: all .2s ease-in;
  transition: all .2s ease-in;
  font-weight: 600;
  font-style: normal;
  min-height: 44px;
  padding: 10px 16px;
  gap: 6px;
  font-size: 16px;
  line-height: 24px;
  color: #CECFD2;
  border: none;
  background: none;
  box-shadow: none;
  border: 1px solid rgba(0,0,0,0);
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:hover:not(:disabled) {
  color: #F0F1F1;
  background: #1F242F;
}

.c0:disabled {
  color: #85888E;
  border: 1px solid rgba(0,0,0,0);
}

<button
  className="c0"
  type="button"
>
  tertiary label
</button>
`;
